src/test/coll.rs

```diff
@@ -753,3 +757,29 @@ async fn no_read_preference_to_standalone() {
 
     assert!(!command_started.command.contains_key("$readPreference"));
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn watch_test() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+    if client.is_standalone() {
+        return;
+    }
+
+    let coll = client
+        .init_db_and_coll(function_name!(), function_name!())
+        .await;
+
+    let pipeline = vec![doc! { "$match": { "x": 3 }}];
+    let options = ChangeStreamOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .max_await_time(Duration::from_millis(5))
+        .batch_size(10)
+        .full_document(FullDocumentType::UpdateLookup)
+        .build();
+    let result = coll.watch(pipeline, Some(options)).await;
+    assert!(result.is_ok());
```

Comment:
Ditto above about unwrapping instead of asserting

Reply:
fixed